app/controllers/admin/orders_controller.rb

```diff
@@ -0,0 +1,25 @@
+class Admin::OrdersController < ApplicationController
+  before_action :require_admin
+
+  def require_admin
+    render file: "/public/404" unless current_admin?
+  end
+
+  def update
+    @order = Order.find(order_params[:order_id])
+    if order_params[:type] == "ship"
```

Comment:
We made the decision to pass an additional param with the patch request to update so that this functionality only executes when we want to ship. Assuming there's potential for different types of updates later which we can handle with different params passed.